src/main/java/org/zone/region/flag/entity/monster/move/MonsterPreventionListener.java

```diff
@@ -36,4 +37,23 @@ public void onMonsterMoveForMonsterPrevention(MoveEntityEvent event,
         }
         event.setCancelled(true);
     }
-}
+
+    @Listener
+    public void onMonsterSpawnEvent(SpawnEntityEvent event, Monster monster) {
+        boolean contains = event
+                .entities()
+                .stream()
+                .filter(entity ->
+                                entity instanceof Monster)
+                .anyMatch(entity -> {
+                 Optional<Zone> opZone = ZonePlugin
+                    .getZonesPlugin()
+                    .getZoneManager()
+                    .getPriorityZone(monster.world(), monster.position());
```

Comment:
This should be ``entity`` not ``monster``

Reply:
why?

Reply:
Because this event doesnt have a Monster in the cause, therefore you should remove the monster parameter. Therefore your left with the entity argument from the ``anyMatch``

Reply:
k